test/tests/functional/pbs_job_task.py

```diff
@@ -55,31 +55,33 @@ def test_singlenode_pbsdsh(self):
         This test case validates that task started by pbsdsh runs
         properly within a single-noded job.
         """
-
-        job = Job(TEST_USER)
+        a = {ATTR_S: '/bin/bash'}
+        job = Job(TEST_USER, attrs=a)
         script = ['pbsdsh echo "OK"']
         job.create_script(body=script)
         jid = self.server.submit(job)
-
         self.server.expect(JOB, {'job_state': 'F'}, id=jid, extend='x')
 
         job_status = self.server.status(JOB, id=jid, extend='x')
         if job_status:
             job_output_file = job_status[0]['Output_Path'].split(':')[1]
 
-        with open(job_output_file, 'r') as fd:
-            job_out = fd.read().strip()
-            self.logger.info("job_out=%s" % (job_out,))
-
-        self.assertEqual(job_out, "OK")
+        ret = self.du.cat(hostname=self.mom.shortname,
+                          filename=job_output_file,
+                          runas=TEST_USER)
+        _msg = "cat command failed with error:%s" % ret['err']
+        self.assertEqual(ret['rc'], 0, _msg)
+        _msg = 'Job\'s error file has error:"%s"' % ret['out']
```

Comment:
Put a space after the colon (here and other places).

Reply:
@bhroam: By default it uses csh shell and the job's output script will have warnings as below
- Warning: no access to tty (Inappropriate ioctl for device).
+ OK
 : Job's error file has error:"['Warning: no access to tty (Inappropriate ioctl for device).', 'Thus no job control in this shell.', '/bin/csh', 'OK']"

So I am pass ATTR_S parameter to avoid this